extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java

```diff
@@ -208,6 +216,64 @@ public KubernetesPortBuildItem kubernetesForManagement(
                 managementBuildTimeConfig.enabled());
     }
 
+    @BuildStep
+    void registerLocalStaticResources(VertxHttpBuildTimeConfig httpBuildTimeConfig,
+            BuildProducer<GeneratedStaticResourceBuildItem> generatedStaticResources) {
+
+        var localStatic = httpBuildTimeConfig.localStaticResources();
+        if (!localStatic.enabled()) {
+            return;
+        }
+
+        final String basePath = localStatic.normalizedPath();
+
+        Path root = Path.of(localStatic.directory()).normalize();
+        if (!Files.isDirectory(root)) {
+            return;
+        }
+
+        try (Stream<Path> paths = Files.walk(root)) {
+            paths.filter(Files::isRegularFile)
+                    .forEach(file -> {
+                        Path relative = root.relativize(file);
+                        String relativeUnix = relative.toString().replace('\\', '/');
+
+                        String endpoint = basePath + "/" + relativeUnix;
+                        if (endpoint.endsWith("/")) {
+                            endpoint = endpoint.substring(0, endpoint.length() - 1);
+                        }
+
+                        generatedStaticResources.produce(
+                                new GeneratedStaticResourceBuildItem(endpoint, file));
+                    });
+        } catch (IOException e) {
+            throw new UncheckedIOException("Failed to register local static resources from directory " + root, e);
+        }
+    }
+
+    @BuildStep
+    void watchLocalStaticResourcesForDev(VertxHttpBuildTimeConfig httpBuildTimeConfig,
+            BuildProducer<HotDeploymentWatchedFileBuildItem> watchedFiles) {
+
+        var localStatic = httpBuildTimeConfig.localStaticResources();
+        if (!localStatic.enabled()) {
+            return;
+        }
+
+        String dir = localStatic.directory();
+        if (dir == null || dir.isBlank()) {
+            return;
+        }
+
+        String normalizedDir = Path.of(dir).normalize().toString();
```

Comment:
you might need `toAbsolutePath` here, because AFAIK the watcher is using absolute path for non resources. Debug to be sure